### PR TITLE
Improvements before migration to .NET8

### PR DESCRIPTION
**Pull request by wcoder**
### Description

- Add `SupportedOSPlatformVersion` for packages
- Update nuspecs
- Sync target frameworks 

### API Changes

 None

### Platforms Affected

- Core (all platforms)
- iOS
- Android

### Behavioral/Visual Changes

None

### Before/After Screenshots

Not applicable

### PR Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Softeq/XToolkit.WhiteLabel/blob/master/.github/CONTRIBUTING.md) document
- [x] My code follows the [code styles](https://github.com/Softeq/dotnet-guidelines/tree/xamarin_guidelines)
- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: nuget/Softeq.XToolkit.Permissions.nuspec [hunk 1]
```diff
@@ -11,18 +11,18 @@
...
     <repository type="git" url="https://github.com/Softeq/XToolkit.WhiteLabel.git" />
     <description>Simple cross platform plugin to request and check permissions for Android and iOS.</description>
-    <copyright>Copyright 2022 Softeq Development Corp.</copyright>
+    <copyright>Copyright 2023 Softeq Development Corp.</copyright>
     <tags>softeq, xtoolkit, xamarin, ios, android, permissions</tags>
     <releaseNotes/>
     <dependencies>
-      <group targetFramework="netstandard2.1">
+      <group targetFramework="net6.0">
         <dependency id="Xamarin.Essentials" version="1.7.3" />
```
Comment (dimysa) [inline nuget/Softeq.XToolkit.Permissions.nuspec:19]: There don't seem to be any dependencies on Xamarin.Essentials anymore

**Comment by wcoder** [comment]
@dimysa you are right about old dependencies in nuspecs but currently, it's out of the scope of this PR
Could you please check again?

**Comment by dimysa** [comment]
> @dimysa you are right about old dependencies in nuspecs but currently, it's out of the scope of this PR Could you please check again?

The PR description says to update nuspecs, so it seems strange to me that this is out of scope. Maybe I'm wrong, but we can't use this nuspec to create nugets now.
But if you say it's out of scope, I assume it will be fully changed in the future?

**Review by dimysa — APPROVED** [review]
In general the changes look clear, but there are questions about nuspecs

**Comment by wcoder** [comment]
> But if you say it's out of scope, I assume it will be fully changed in the future?
Yes you are right, nuspecs approach will be reworked in future

**Pull request merged**